sherpa/conftest.py

```diff
@@ -83,17 +83,32 @@ def pytest_addoption(parser):
     parser.addoption("--runzenodo", action="store_true", default=False,
                      help="run tests that query Zenodo (requires internet)")
 
+    parser.addoption("--skipds9", action="store_false", default=True,
+                     help="skip DS9 tests if DS9 is installed")
+
 
 def pytest_collection_modifyitems(config, items):
 
     # Skip tests unless --runxxx given in cli
     #
     for label in ["slow", "zenodo"]:
-        opt = "--run{}".format(label)
+        opt = f"--run{label}"
+        if config.getoption(opt):
+            continue
+
+        skip = pytest.mark.skip(reason=f"need {opt} option to run")
+        for item in items:
+            if label in item.keywords:
+                item.add_marker(skip)
+
+    # Skip tests if --skipxxx given in cli
+    #
+    for label in ["ds9"]:
+        opt = f"--skip{label}"
         if config.getoption(opt):
```

Comment:
Maybe I'm just a bit stupid, but should this not be the inverse? If "skipds9" is set, then we continue, but if it is not set, then `pytest.mark.skip(reason=f"{opt} was set")`? That looks the same as above for the "runxxx" tests and I would have expected a logical not (`~`) somewhere in this section. Or do I need to start drinking coffee?

Reply:
However, we don't have to spend time to understand this. If it works, it works; let's be efficient and get this in instead of waiting for one of us to make perfectly engeneered code for something that's such a special case about just invoking tests.